### DISCUSSION TITLE
algebra relazionale - 1

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/basi-di-dati-1#58] [Esercizi]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/15af1ff0-4e5e-4ff4-92ee-d539cc320c6d)

**Comment by 7Andrei** [comment]
![AlgebraRelazionale1](https://github.com/sapienzastudentsnetwork/bdi2324/assets/97444723/66239b7a-f6b6-4ab8-9e03-17f96fd508e2)

**Comment by Carlo-Armato** [comment]
![1](https://github.com/sapienzastudentsnetwork/bdi2324/assets/158755407/976b143a-ce56-47ed-ad69-fc67fc905535)

**Comment by FeddyLix17** [comment]
Date le relazioni

<table align="left">
    <tr>
        <th colspan="3"> PRODOTTI </th>
    </tr>
    <tr>
        <td> $\underline{\textnormal{Codice}}$ </td>
        <td> Descrizione </td>
        <td> Prezzo_Unit </td>
    </tr>
    <tr>
        <td> $\dots$ </td>
        <td>$\dots$ </td>
        <td> $\dots$ </td>
    </tr>
    <tr>
        <td> $\dots$ </td>
        <td>$\dots$ </td>
        <td> $\dots$ </td>
    </tr>
</table>

<table align="right">
    <tr>
        <th colspan="3"> MAGAZZINI </th>
    </tr>
    <tr>
        <td> $\underline{\textnormal{Codice}}$ </td>
        <td> Indirizzo </td>
        <td> Città </td>
    </tr>
    <tr>
        <td> $\dots$ </td>
        <td>$\dots$ </td>
        <td> $\dots$ </td>
    </tr>
    <tr>
        <td> $\dots$ </td>
        <td>$\dots$ </td>
        <td> $\dots$ </td>
    </tr>
</table>

<table align="center">
    <tr>
        <th colspan="3"> SCORTE </th>
    </tr>
    <tr>
        <td> $\underline{\textnormal{Codice-P}}$ </td>
        <td>$\underline{\textnormal{Codice-M}}$ </td>
        <td> N-pezzi </td>
    </tr>
    <tr>
        <td> $\dots$ </td>
        <td>$\dots$ </td>
        <td> $\dots$ </td>
    </tr>
    <tr>
        <td> $\dots$ </td>
        <td>$\dots$ </td>
        <td> $\dots$ </td>
    </tr>
</table>

dove

- _MAGAZZINI.Codice_ è la chiave primaria di MAGAZZINI

- _PRODOTTI.Codice_ è la chiave primaria di PRODOTTI

- _SCORTE.Codice-P References PRODOTTI.Codice_

- _SCORTE.Codice-M References MAGAZZINI.Codice_ 

## a)  Dati dei magazzini che hanno una scorta di almeno 10 frigoriferi

Bisognerà dunque 

- selezionare i dati dei magazzini, $\Large \pi_{Codice,\ Indirizzo,\ Città}$

- dal _Theta Join_ fra le 3 relazioni (nel quale i rispettivi codici coincidano)

$$
    \Large  MAGAZZINI \bowtie_{\ Codice\ =\ Codice-M}(PRODOTTI  \bowtie_{\ Codice\ =\ Codice-P}SCORTE)
$$

- dove il prodotto è un frigorifero con almeno 10 scorte presenti, $\Large \sigma_{Descrizione\ =\ \text{"Frigorifero"}\ \wedge\ N-pezzi\  \geq\ 10}$

$$
    \Large \pi_{Codice,\ Indirizzo,\ Città}\ ( 
$$

$$
    \Large \sigma_{Descrizione\ =\ \text{"Frigorifero"}\ \wedge\ N-pezzi\  \geq\ 10}\ ( 
$$

$$
    \Large MAGAZZINI \bowtie_{\ Codice\ =\ Codice-M}(PRODOTTI  \bowtie_{\ Codice\ =\ Codice-P}SCORTE)\ )\ )
$$

## b)  Dati dei prodotti per cui non ci sono scorte in alcun magazzino di Roma

In questo caso, invece, bisognerà 

- selezionare i dati dei prodotti, $\Large \pi_{Codice,\ Descrizione,\ Prezzo \textunderscore unit}$

- dal _Theta Join_ fra le 3 relazioni (nel quale i rispettivi codici coincidano)

$$
    \Large  MAGAZZINI \bowtie_{\ Codice\ =\ Codice-M}(PRODOTTI  \bowtie_{\ Codice\ =\ Codice-P}SCORTE)
$$

- dove il magazzino sia a roma e non ci siano scorte presenti dato un prodotto, $\Large \sigma_{Città\ =\ \text{"Roma"}\ \wedge\ N-pezzi\  =\ 0}$

$$
    \Large \pi_{Codice,\ Descrizione,\ Prezzo \textunderscore unit}\ ( 
$$

$$
    \Large  \sigma_{Città\ =\ \text{"Roma"}\ \wedge\ N-pezzi\  =\ 0}\ ( 
$$

$$
    \Large MAGAZZINI \bowtie_{\ Codice\ =\ Codice-M}(PRODOTTI  \bowtie_{\ Codice\ =\ Codice-P}SCORTE)\ )\ )
$$

**Comment by CiottoloMaggico** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/79638284/61d7a0bb-f5f0-4d2c-bc28-cde11515f54d)
